### PR TITLE
v1 of xauth support

### PR DESCRIPTION
**Pull request by aren**
rough draft; might not work out of the box. still needs a lot of testing.

also needing attention (at least): /etc/sysctl.conf

### CODE REVIEW

File: templates/ipsec.conf [hunk 1]
```diff
@@ -2,28 +2,50 @@ version 2.0
 
 config setup
   dumpdir=/var/run/pluto/
-  virtual_private=%v4:10.0.0.0/8,%v4:192.168.0.0/16,%v4:172.16.0.0/12,%v4:!<LOCAL_SUBNET>
+  virtual-private=%v4:10.0.0.0/8,%v4:192.168.0.0/16,%v4:172.16.0.0/12,%v4:!<LT2P_CIDR>,%v4:!<XAUTH_CIDR>
```
Comment (ttheune) [inline templates/ipsec.conf]: does the `-` vs `_` matter here?

File: static/pluto [hunk 2]
```diff
@@ -0,0 +1,3 @@
+auth		required	pam_radius_auth.so
+account		required	pam_radius_auth.so
```
Comment (ttheune) [inline static/pluto]: whitespace funkyness

File: templates/ipsec.conf (hunk 1 above)
Comment (aren) [reply templates/ipsec.conf]: Docs definitely say "virtual-private" today. Wonder if it supported both in the past? Or if this just never worked?

https://libreswan.org/man/ipsec.conf.5.html

File: static/pluto (hunk 2 above)
Comment (aren) [reply static/pluto]: weird, since vim has 8 spaces per tab, and that's also the github viewer default. oh well, i converted the file to spaces.

**Comment by ttheune** [comment]
Looks good at this point.  Do we want to keep testing this, or work on incorporating the docker build you found from hwdsl2?
(https://hub.docker.com/r/hwdsl2/ipsec-vpn-server)

**Comment by aren** [comment]
I'd like to get this working since it's a direct replacement for the current VPN. Then we can focus on the docker container.

**Comment by ttheune** [comment]
AMI builds cleanly enough.  Let's move it to regular PR.

**Comment by ttheune** [comment]
Test host: 52.11.232.124
FoxFood-ified

**Comment by danishfoxpass** [comment]
@ttheune Travis, is the plan to make progress and merge this before Monday? Been a long while on this.

**Comment by ttheune** [comment]
This needs a bit of re-working now with some of the changes we've built into the main branch.  Testing continues.

**Comment by danishfoxpass** [comment]
@ttheune any comments on how to proceed on this?

**Comment by ttheune** [comment]
My latest changes need to be reviewed/approved

**Comment by danishfoxpass** [comment]
@ttheune sure, can you pls resolve branch conflicts.

**Comment by danishfoxpass** [comment]
@aren can you review this?

**Comment by aren** [comment]
I approve the additional changes. I can't merge as it's my PR.

**Comment by danishfoxpass** [comment]
Ok merged it.

**Pull request merged**